### DISCUSSION TITLE
Environments - Problem an answer is desired

### DISCUSSION THREAD

**Discussion by GreimersStefan** [mbnuqw/sidebery#938] [General]
## Description
Hello Max,
You had already seen a video on t-online.de from the preferences and tabs which weren´t sychronized. Now I have tooken a bit further with Tor and with the Tor Project were I have contact to. Extentsions come all from firefox mozilla. Speaking to you, it belongs to environments in sideberry. I have serveral environments (which are always empty). The intension is about to get the domain-names into the environments, afterwards to show them in the corresponding environment. Such as TELEKOM > https://accounts.login.idm.telekom.com/oauth2/auth?client_id=10LIVESAM30000004901PORTAL00000000000000&state=473fbfdc18ff672ceb80c6e081d06818ebfb52745963da0d2cb8b4d873917e05&claims=%7B%22id_token%22%3A%7B%22urn%3Atelekom.com%3Aall%22%3Anull%7D%7D&nonce=473fbfdc18ff672ceb80c6e081d06818ebfb52745963da0d2cb8b4d873917e05&redirect_uri=https%3A%2F%2Flogin.t-online.de%2Fcallback&logout_uri=https%3A%2F%2Flogin.t-online.de%2Ftelekom_logout&display=popup&scope=openid&response_type=code

and second http://kabelspeed.telekom-dienste.de/.

But the environments are always empty. With a mouse gesture goes the domain name into the environment, but it did not stay there. I have already seen, that your software sideberry did not show anymore the environments in the sidebar, though its are in Tor Preferences and you Menue Options in Sidebar not always the same and at the copying occurrence is done but not safed.

Tor Browser has get the domain names of Opera by export of a html-file converted with Visual Studio Code in .js. When there is a problem with domain names, I could use another file format to save an import for sideberry like .jsx .json or a little script with Apple Scripteditor.app.
We wait in the meantime for the mozilla extension update.
When you can, please give me an address of your donate account.
Thank you, Stefan Greimers

**Comment by mbnuqw** [comment]
If by environments you mean containers - Firefox (and TorBrowser) doesn't allow using containers in private window. And since by default TorBrowser uses private windows, you can't open/reopen tabs in a different (non-private) container. You need to open TorBrowser settings, go to the "Privacy and Security", then toggle off the "Always use private browser mode" option. After that, you will be able to use containers, bind URLs to them and so on...

Also, you can use Tor network in standard Firefox. To do that, you need to install Tor service and find out the IP:PORT to connect to it as a SOCKS5 proxy. Then, open sidebery settings, create "Tor" container and configure Tor proxy for it (IP and PORT might be different):
![screenshot-2023-02-12 14:46:03](https://user-images.githubusercontent.com/6276694/218303895-6ddc841a-51da-4b41-afba-8000b1acb0b4.png)

If I didn't understand your question correctly, please, provide step-by-step description of what are you trying to do and what is the expected behavior and what is the actual behavior.

Moving to discussions as a question.

**Reply by GreimersStefan** [reply]
Max Badryzlov, first thank you for your work. What you have sent to Tor Browser as Screenshot is not in the content of Tor Browser.  Your software can in Tor Browser only create icons with colors as intentions. These are than in Tor Browser but URls that are defined in your software go not solid in your program, so that they are in Tor Browser - only in the seach field they may be appear afterwards they were used on a connection. I do not see the URLs in the intention of Sideberry solid after restarts. - That´s it to sideberry on macos+TorBrowser. I have an request to you belonging macOS 11+ Safari. In Apple`s Mac App Store there is nobody who takes care of Bookmarks. You have the package to Mozilla Firefox, but Apple? I know that is differcult as before. Others have for example AntiCaptcha brought in the App Store, which connect to the Account on AntiCaptcha. It runs. When you have an app in preparation for a Apple Safari conclusion it would help to all.

**Comment by GreimersStefan** [comment]
They recommended Firefox for me to download, but there are a few things that are worthwhile compared to Tor Browser. 1. Firefox has columns of numbers when downloading, which also cause confusion in directories such as /usr/bin/ etc. in the Apple operating system. I will not do that. Firefox is a Windows distillation, which means it is fully compatible with public Windows to work without system maintenance. 2. Tor Browser has the possibility to use bridges without these requests or without installing them beforehand as opfs4, Snowflake or meek-azure. This means that if a server connection is not established, another IP address can be used immediately via a button (in the open Internet without your own server) and the second connection is usually established immediately. Firefox can't do that. I have a connection to the Tor Browser project as a co-moderator and see that js style sheets are also being implemented for Tor Browser, which have so far prevented banks and purchase accounts from possibly not or not completely returning the server login data. If this was updated to Tor Browser Project in a new browser macOS update package, Tor Browser would all connect like other browsers and Tor would be 100% compatible. With patience, full functionality will come to this platform as well. - You can only get Java Script on macOS 11 via Terminal on the computer. You've seen your screenshot of "Tor". Finally: I'm exclusively on macOS, I'm only using a Windows evaluation version Win 10 for testing and I'm disappointed after every system start, how Internet connections that weren't wanted could be established immediately without being checked. The many open IP addresses that can be rejected via a VPS server and configured with Wireguard protocols bother me in terms of external perception. But that's rambling. I hope you understand. Thanks.
